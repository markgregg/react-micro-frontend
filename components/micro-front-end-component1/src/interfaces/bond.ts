export type Bond = {
  isinCode: string
  haircutCategory: string,
  type: string,
  refMarket: string,
  demonination: string,
  issuranceDate: string,
  maturityDate: string,
  issuerCsd: string,
  couponRate: number
  issuerName: string,
  issuerResidence: string,
  issuerGroup: string,
  guarantorName: string,
  guarantorResidence: string,
  guarantorGroup: string,
  couponDef: string,
  haircut: number,
  haircutOwn: number,
  custom: string
};