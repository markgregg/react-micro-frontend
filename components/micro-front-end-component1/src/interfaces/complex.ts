export interface Complex {
  name: string;
  value: number;
  description: string;
}
