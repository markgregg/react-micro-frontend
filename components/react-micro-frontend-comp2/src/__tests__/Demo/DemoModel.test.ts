import { 
  createDemoModel, 
  formatDate, 
  stringToDate, 
  stringToIsoDateString,
  dateCompare
} from "../../Demo/DemoModel"
import { Bond } from "../../interfaces/bond";

test("format date", () => {
  const dateString = formatDate(new Date(2022, 6, 10));

  expect(dateString).toBe("10/6/2022");
});

test("string to date", () => {
  const date = stringToDate("22/02/2021")

  expect(date).toStrictEqual(new Date(2021, 1, 22));
});

test("string to ISO date string", () => {
  const dateString = stringToIsoDateString("22/02/2021")

  expect(dateString).toStrictEqual("2021-02-22T00:00:00.000Z");
});

describe("date compare functions", () => {

  test("compre date not string", () => {
    const result = dateCompare(new Date(Date.now()), undefined);
    expect(result).toBe(0);
  })

  test("compre date less", () => {
    const result = dateCompare(new Date(2022,4,10), "01/04/2022");
    expect(result).toBe(-1);
  })

  test("compre date greater", () => {
    const result = dateCompare(new Date(2022,4,10), "01/04/2023");
    expect(result).toBe(1);
  })

  test("compre date equal", () => {
    const result = dateCompare(new Date(2022,6,10), "10/07/2022");
    expect(result).toBe(0);
  })

});

describe("model functions", () => {
  test("create model", () => {
    const model = createDemoModel([]);
  
    expect(JSON.stringify(model)).toMatchSnapshot();
  });

  test("uniqueValues", () => {
    const bondList: Bond[] = [
      {"isinCode":"FR0013405222","haircutCategory":"L1D","type":"AT01","refMarket":"RMFR01","demonination":"USD","issuranceDate":"04/03/2019","maturityDate":"04/03/2024","issuerCsd":"CLFR01","couponRate":5.655,"issuerName":"BPCE","issuerResidence":"IRFR","issuerGroup":"IG4","guarantorName":"","guarantorResidence":"","guarantorGroup":"","couponDef":"CD2","haircut":21.7,"haircutOwn":0,"custom":"N"},
      {"isinCode":"FR0127739573","haircutCategory":"L1D","type":"AT03","refMarket":"RMFR04","demonination":"USD","issuranceDate":"01/12/2022","maturityDate":"01/12/2023","issuerCsd":"CLFR01","couponRate":0,"issuerName":"BPCE","issuerResidence":"IRFR","issuerGroup":"IG4","guarantorName":"","guarantorResidence":"","guarantorGroup":"","couponDef":"CD1","haircut":21.7,"haircutOwn":0,"custom":"N"},
      {"isinCode":"FR0013213683","haircutCategory":"L1C","type":"AT01","refMarket":"RMFR01","demonination":"EUR","issuranceDate":"19/10/2016","maturityDate":"19/10/2028","issuerCsd":"CLFR01","couponRate":0.95,"issuerName":"SOCIETE DES AUTOROUTES DU NORD ET DE L EST DE LA FRANCE (SANEF)","issuerResidence":"IRFR","issuerGroup":"IG3","guarantorName":"","guarantorResidence":"","guarantorGroup":"","couponDef":"CD4","haircut":16.7,"haircutOwn":0,"custom":"N"}
    ];
    const model = createDemoModel(bondList);
  
    expect(JSON.stringify(model.choices[1].options)).toBe("[\"L1C\",\"L1D\"]");
  });

  test("isinLlookUp", async () => {
    const bondList: Bond[] = [
      {"isinCode":"FR0013405222","haircutCategory":"L1D","type":"AT01","refMarket":"RMFR01","demonination":"USD","issuranceDate":"04/03/2019","maturityDate":"04/03/2024","issuerCsd":"CLFR01","couponRate":5.655,"issuerName":"BPCE","issuerResidence":"IRFR","issuerGroup":"IG4","guarantorName":"","guarantorResidence":"","guarantorGroup":"","couponDef":"CD2","haircut":21.7,"haircutOwn":0,"custom":"N"},
      {"isinCode":"FR0127739573","haircutCategory":"L1D","type":"AT03","refMarket":"RMFR04","demonination":"USD","issuranceDate":"01/12/2022","maturityDate":"01/12/2023","issuerCsd":"CLFR01","couponRate":0,"issuerName":"BPCE","issuerResidence":"IRFR","issuerGroup":"IG4","guarantorName":"","guarantorResidence":"","guarantorGroup":"","couponDef":"CD1","haircut":21.7,"haircutOwn":0,"custom":"N"},
      {"isinCode":"FR0013213683","haircutCategory":"L1C","type":"AT01","refMarket":"RMFR01","demonination":"EUR","issuranceDate":"19/10/2016","maturityDate":"19/10/2028","issuerCsd":"CLFR01","couponRate":0.95,"issuerName":"SOCIETE DES AUTOROUTES DU NORD ET DE L EST DE LA FRANCE (SANEF)","issuerResidence":"IRFR","issuerGroup":"IG3","guarantorName":"","guarantorResidence":"","guarantorGroup":"","couponDef":"CD4","haircut":16.7,"haircutOwn":0,"custom":"N"}
    ];
    const model = createDemoModel(bondList);
  
    const result =  await model.choices[0].lookUp!!("1277");
    
    console.log(result);
    expect(result).toStrictEqual(["FR0127739573"]);
  });

});


