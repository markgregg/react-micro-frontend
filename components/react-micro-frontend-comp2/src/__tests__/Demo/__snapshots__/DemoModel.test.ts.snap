// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create model 1`] = `"{"choices":[{"key":"isinCode","prefix":"Isin","agGridColumn":"isinCode","choiceDisplayName":"Isin","regExMatch":{}},{"key":"haircutCategory","prefix":"Haircut Cut","options":[],"agGridColumn":"haircutCategory","choiceDisplayName":"Haircut Cut"},{"key":"type","prefix":"type","options":[],"agGridColumn":"type","choiceDisplayName":"Type"},{"key":"refMarket","prefix":"ref mkt","options":[],"agGridColumn":"refMarket","choiceDisplayName":"Reference Market"},{"key":"demonination","prefix":"demon","options":[],"agGridColumn":"demonination","choiceDisplayName":"Demonination"},{"key":"coupon","prefix":"coupon rate","agGridColumn":"couponRate","choiceDisplayName":"Coupon Rate","regExMatch":{},"filterType":"number","replaceExisting":true},{"key":"issuranceDate","prefix":"issurance date","choiceDisplayName":"Issurance Date","regExMatch":{},"agGridColumn":"issuranceDate","filterType":"date","replaceExisting":true},{"key":"maturityDate","prefix":"maturity date","choiceDisplayName":"Maturity Date","regExMatch":{},"agGridColumn":"maturityDate","filterType":"date","replaceExisting":true},{"key":"issuerCsd","prefix":"csd","options":[],"agGridColumn":"issuerCsd","choiceDisplayName":"Issuer CSD"},{"key":"issuerName","prefix":"issuerName","choiceDisplayName":"Issuer Name","agGridColumn":"issuerName","regExMatch":{},"replaceExisting":true},{"key":"issuerResidence","prefix":"issuer res","options":[],"agGridColumn":"issuerResidence","choiceDisplayName":"Issuser Residence"},{"key":"issuerGroup","prefix":"issuer grp","options":[],"agGridColumn":"issuerGroup","choiceDisplayName":"Issuer Group"},{"key":"guarantorName","prefix":"guarantorName","choiceDisplayName":"Guarantor Name","agGridColumn":"guarantorName","regExMatch":{},"replaceExisting":true},{"key":"guarantorResidence","prefix":"guarantor res","options":[],"agGridColumn":"guarantorResidence","choiceDisplayName":"Guarantor Residence"},{"key":"guarantorGroup","prefix":"guarantor grp","options":[],"agGridColumn":"guarantorGroup","choiceDisplayName":"Guarantor Group"},{"key":"couponDef","prefix":"coupon def","options":[],"agGridColumn":"couponDef","choiceDisplayName":"Coupon Definition"},{"key":"haircut","prefix":"haircut","choiceDisplayName":"Haircut","agGridColumn":"haircut","regExMatch":{},"filterType":"number","replaceExisting":true}],"columnDefs":[{"field":"isinCode","headerName":"ISIN Code","filter":"agSetColumnFilter","resizable":true,"width":150},{"field":"haircutCategory","headerName":"Haircut Category","filter":"agSetColumnFilter","resizable":true,"width":120},{"field":"type","headerName":"Type","filter":"agSetColumnFilter","resizable":true,"width":90},{"field":"refMarket","headerName":"Reference","filter":"agSetColumnFilter","resizable":true,"width":120},{"field":"demonination","headerName":"Demonination","filter":"agSetColumnFilter","resizable":true,"width":140},{"field":"issuranceDate","headerName":"Issuance Date","filter":"agDateColumnFilter","filterParams":{},"resizable":true,"width":130},{"field":"maturityDate","headerName":"Maturity Date","filter":"agDateColumnFilter","filterParams":{},"resizable":true,"width":130},{"field":"issuerCsd","headerName":"Issuer CSD","filter":"agSetColumnFilter","resizable":true,"width":120},{"field":"couponRate","headerName":"Coupon Rate","filter":"agNumberColumnFilter","resizable":true,"width":140},{"field":"issuerName","headerName":"Issuer Name","filter":"agTextColumnFilter","resizable":true,"width":300},{"field":"issuerResidence","headerName":"Issuer Residence","filter":"agSetColumnFilter","resizable":true,"width":130},{"field":"issuerGroup","headerName":"Issuer Group","filter":"agSetColumnFilter","resizable":true,"width":130},{"field":"guarantorName","headerName":"Guarantor Name","filter":"agTextColumnFilter","resizable":true,"width":300},{"field":"guarantorResidence","headerName":"Guarantor Residence","filter":"agSetColumnFilter","resizable":true,"width":130},{"field":"guarantorGroup","headerName":"Guarantor Group","filter":"agSetColumnFilter","resizable":true,"width":130},{"field":"couponDef","headerName":"Coupon Definition","filter":"agSetColumnFilter","resizable":true,"width":130},{"field":"haircut","headerName":"Haircut","filter":"agNumberColumnFilter","resizable":true,"width":150}],"rowData":[]}"`;

exports[`model functions create model 1`] = `"{"choices":[{"key":"isinCode","prefix":"Isin","agGridColumn":"isinCode","choiceDisplayName":"Isin","regExMatch":{}},{"key":"haircutCategory","prefix":"Haircut Cut","options":[],"agGridColumn":"haircutCategory","choiceDisplayName":"Haircut Cut"},{"key":"type","prefix":"type","options":[],"agGridColumn":"type","choiceDisplayName":"Type"},{"key":"refMarket","prefix":"ref mkt","options":[],"agGridColumn":"refMarket","choiceDisplayName":"Reference Market"},{"key":"demonination","prefix":"demon","options":[],"agGridColumn":"demonination","choiceDisplayName":"Demonination"},{"key":"coupon","prefix":"coupon rate","agGridColumn":"couponRate","choiceDisplayName":"Coupon Rate","regExMatch":{},"filterType":"number","replaceExisting":true},{"key":"issuranceDate","prefix":"issurance date","choiceDisplayName":"Issurance Date","regExMatch":{},"agGridColumn":"issuranceDate","filterType":"date","replaceExisting":true},{"key":"maturityDate","prefix":"maturity date","choiceDisplayName":"Maturity Date","regExMatch":{},"agGridColumn":"maturityDate","filterType":"date","replaceExisting":true},{"key":"issuerCsd","prefix":"csd","options":[],"agGridColumn":"issuerCsd","choiceDisplayName":"Issuer CSD"},{"key":"issuerName","prefix":"issuerName","choiceDisplayName":"Issuer Name","agGridColumn":"issuerName","regExMatch":{},"replaceExisting":true},{"key":"issuerResidence","prefix":"issuer res","options":[],"agGridColumn":"issuerResidence","choiceDisplayName":"Issuser Residence"},{"key":"issuerGroup","prefix":"issuer grp","options":[],"agGridColumn":"issuerGroup","choiceDisplayName":"Issuer Group"},{"key":"guarantorName","prefix":"guarantorName","choiceDisplayName":"Guarantor Name","agGridColumn":"guarantorName","regExMatch":{},"replaceExisting":true},{"key":"guarantorResidence","prefix":"guarantor res","options":[],"agGridColumn":"guarantorResidence","choiceDisplayName":"Guarantor Residence"},{"key":"guarantorGroup","prefix":"guarantor grp","options":[],"agGridColumn":"guarantorGroup","choiceDisplayName":"Guarantor Group"},{"key":"couponDef","prefix":"coupon def","options":[],"agGridColumn":"couponDef","choiceDisplayName":"Coupon Definition"},{"key":"haircut","prefix":"haircut","choiceDisplayName":"Haircut","agGridColumn":"haircut","regExMatch":{},"filterType":"number","replaceExisting":true}],"columnDefs":[{"field":"isinCode","headerName":"ISIN Code","filter":"agSetColumnFilter","resizable":true,"width":150},{"field":"haircutCategory","headerName":"Haircut Category","filter":"agSetColumnFilter","resizable":true,"width":120},{"field":"type","headerName":"Type","filter":"agSetColumnFilter","resizable":true,"width":90},{"field":"refMarket","headerName":"Reference","filter":"agSetColumnFilter","resizable":true,"width":120},{"field":"demonination","headerName":"Demonination","filter":"agSetColumnFilter","resizable":true,"width":140},{"field":"issuranceDate","headerName":"Issuance Date","filter":"agDateColumnFilter","filterParams":{},"resizable":true,"width":130},{"field":"maturityDate","headerName":"Maturity Date","filter":"agDateColumnFilter","filterParams":{},"resizable":true,"width":130},{"field":"issuerCsd","headerName":"Issuer CSD","filter":"agSetColumnFilter","resizable":true,"width":120},{"field":"couponRate","headerName":"Coupon Rate","filter":"agNumberColumnFilter","resizable":true,"width":140},{"field":"issuerName","headerName":"Issuer Name","filter":"agTextColumnFilter","resizable":true,"width":300},{"field":"issuerResidence","headerName":"Issuer Residence","filter":"agSetColumnFilter","resizable":true,"width":130},{"field":"issuerGroup","headerName":"Issuer Group","filter":"agSetColumnFilter","resizable":true,"width":130},{"field":"guarantorName","headerName":"Guarantor Name","filter":"agTextColumnFilter","resizable":true,"width":300},{"field":"guarantorResidence","headerName":"Guarantor Residence","filter":"agSetColumnFilter","resizable":true,"width":130},{"field":"guarantorGroup","headerName":"Guarantor Group","filter":"agSetColumnFilter","resizable":true,"width":130},{"field":"couponDef","headerName":"Coupon Definition","filter":"agSetColumnFilter","resizable":true,"width":130},{"field":"haircut","headerName":"Haircut","filter":"agNumberColumnFilter","resizable":true,"width":150}],"rowData":[]}"`;

exports[`model functions uniqueValues 1`] = `"["L1C","L1D"]"`;
